### PR TITLE
PRC-8 : Flyway migrations - basic

### PR DESCRIPTION
**Pull request by prabash-moj**
[PRC-8 ](https://dsdmoj.atlassian.net/browse/PRC-8): 

**Changes:** 

1. Add  initial configuration required for running flyway migration
2. Add a dummy baseline flyway script for testing the functionality- This will be replaced later when the data model is ready
3. Prisoner search client api client code generated by the open api so it can consume data from prisoner search api 

**NOTE**: _Remove /attribute-search BETA search api configuration from  the Prisoner search api as there is limitation on OpenApi tool to generate polymorphic types,_

Test evidence :
1) App starting with some dummy flyway scripts : 
image

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,6 +7,10 @@ WORKDIR /app
 ADD . .
 RUN ./gradlew --no-daemon assemble
 
+# Grab AWS RDS Root cert
+RUN apt-get update && apt-get install -y curl
+RUN curl https://s3.amazonaws.com/rds-downloads/rds-ca-2019-root.pem  > root.crt
+
```
Comment (timharrison-moj) [inline Dockerfile:13]: Think there's a more recent root cert this year? I remember updating this one in BVLS Api recently.

Comment (timharrison-moj) [reply Dockerfile:13]: Check current source please?

File: README.md [hunk 2]
```diff
@@ -1,49 +1,72 @@
...
+To test Gov Notify emails locally, you just need to add one more variable to your `.env` file.
+
+```
+export NOTIFY_API_KEY=<gov.notify.api.key>
+```
+If you have added it correctly, you will see the log on startup with the following output:
+
+```
+Gov Notify emails are enabled
+```
```
Comment (timharrison-moj) [inline README.md:72]: Thanks for updating the README.

File: run-local.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+#
+
+# Provide the DB connection details to local container-hosted Postgresql DB already running
+export DB_SERVER=localhost
+export DB_NAME=contacts-db
+export DB_USER=contacts
+export DB_PASS=contacts
+export DB_SSL_MODE=prefer
+export $(cat .env | xargs)  # If you want to set or update the current shell environment e.g. system client and secret.
+
```
Comment (timharrison-moj) [inline run-local.sh:18]: Good - no inclusion of sensitive secrets (client id/secret)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/client/prisonersearch/PrisonerSearchClient.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  fun getPrisoner(prisonerNumber: String): Prisoner? =
+    prisonerSearchApiWebClient
+      .get()
+      .uri("/prisoner/{prisonerNumber}", prisonerNumber)
+      .retrieve()
+      .bodyToMono(Prisoner::class.java)
+      .onErrorResume(WebClientResponseException.NotFound::class.java) { Mono.empty() }
+      .block()
+}
+
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/client/prisonersearch/PrisonerSearchClient.kt:18]: This is the API we will probably use most - to get the details of prisoners / groups of prisoners.

File: openapi-specs/prison-api.json [hunk 5]
```diff
@@ -0,0 +1 @@
...th","description":"The prison id","required":true,"schema":{"type":"string"}},{"name":"movementDate","in":"query","description":"Optional filter on date of movement. Defaults to today","required":false,"schema":{"type":"string","format":"date"}}],"responses":{"500":{"description":"Unrecoverable error occurred whilst processing request.","content":{"application/json":{"schema":{"$ref":"#/components/schemas/ErrorResponse"}}}},"200":{"description":"OK","content":{"application/json":{"schema":{"type":"integer","format":"int32"}}}},"400":{"description":"Invalid request.","content":{"application/json":{"schema":{"$ref":"#/components/schemas/ErrorResponse"}}}}}}},"/api/movements/livingUnit/{livingUnitId}/currently-out":{"get":{"tags":["movements"],"summary":"Information on offenders currently out.","description":"Requires role ESTABLISHMENT_ROLL.<p>This endpoint uses the REPLICA database.</p>","operationId":"getOffendersCurrentlyOut","parameters":[{"name":"livingUnitId","in":"path","description":"The identifier of a living unit, otherwise known as an internal location.","required":true,"schema":{"type":"integer","format":"int64"}}],"responses":{"500":{"description":"Unrecoverable error oc
```
Comment (timharrison-moj) [inline openapi-specs/prison-api.json:1]: Should this be the openapi spec for PrisonerSearchApi, rather than prisonApi?

**Review by timharrison-moj — APPROVED** [review]
Looking good so far. 
Suggest adding a test table - and some data - and a quick test which verifies it is present? But could wait until we have the first tables of the new data model ready?

File: openapi-specs/prison-api.json (hunk 5 above)
Comment (prabash-moj) [reply openapi-specs/prison-api.json:1]: the documentation says it is prison api 
image

File: src/test/resources/application-test.yml [hunk 6]
```diff
@@ -4,7 +4,7 @@ spring:
     show-sql: true
 
   datasource:
-    url: 'jdbc:h2:mem:book-a-video-link-db;MODE=PostgreSQL'
+    url: 'jdbc:h2:mem:contacts-db;MODE=PostgreSQL'
     username: contacts
```
Comment (timharrison-moj) [inline src/test/resources/application-test.yml:8]: This would do it!

**Review by timharrison-moj — APPROVED** [review]
LGTM

File: Dockerfile (hunk 1 above)
Comment (prabash-moj) [reply Dockerfile:13]: updated

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/client/prisonersearch/PrisonerSearchClient.kt (hunk 4 above)
Comment (prabash-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/client/prisonersearch/PrisonerSearchClient.kt:18]: I will take a look to in the VL project and see if I can get a api resource added to call this external resource and test it locally, but this PR can go without this .

File: openapi-specs/prison-api.json (hunk 5 above)
Comment (timharrison-moj) [reply openapi-specs/prison-api.json:1]: Yes - there are two different APIs - Prison API (a huge, direct into NOMIS API), and PrisonerSearchApi (a smaller, quicker API into elastic search for prisoner details).  We want to use prisoner search API, not prison API.

**Review by timharrison-moj — COMMENTED** [review]
LGTM, though two things:
* I don't seen any Flyway migration files in the PR?
* The naming of migration files is very specific - take a look at other projects - V__<date>_<vers>_<description>.sql or R__<description>.sql (repeatable migrations)
* I think we need prisoner search API types, not prison API types.

File: openapi-specs/prison-api.json (hunk 5 above)
Comment (prabash-moj) [reply openapi-specs/prison-api.json:1]: I have created a ticket for this so I can work separate to this ticket to fix the kotlin compilation error due to openapi generated code,

**Review by timharrison-moj — APPROVED** [review]
LGTM. Go for it. 
We'll replace the tables with real ones when we have a model to work with.

**Pull request merged**